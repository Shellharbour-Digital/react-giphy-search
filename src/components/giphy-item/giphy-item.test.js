import { shallow } from 'enzyme';
import React from 'react';
import GiphyItem from './giphy-item.component';
import 'jest-styled-components';

describe('GiphyItem component', () => {
  const mockProps = {
    img: 'mock.png',
    alt: 'mock image',
    url: 'www.mockurl.com',
    height: 200
  }

  let wrapper = shallow(<GiphyItem {...mockProps} />);

  it('expect to render GiphyItem component', ()  => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect LoadingContainer height to be props.height value', () => {
    expect(wrapper.find('LoadingContainer').prop('height')).toEqual(200);
    expect(wrapper.find('LoadingContainer')).toHaveStyleRule('height', '200px');
  });

  it('correctly sets loading state to false when GifImg has loaded', () => {
    wrapper.find('GifImg').props().onLoad();
    expect(wrapper.state()).toEqual({ loading: false });
  });
});
