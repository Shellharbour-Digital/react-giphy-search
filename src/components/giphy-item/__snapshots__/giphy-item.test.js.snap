// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyItem component expect to render GiphyItem component 1`] = `
<styled.div>
  <LoadingContainer
    height={200}
  >
    <Spinner />
  </LoadingContainer>
  <styled.a
    href="www.mockurl.com"
    target="_blank"
  >
    <GifImg
      alt="mock image"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    />
  </styled.a>
</styled.div>
`;
